### PR TITLE
feat(scaffold): add prompts for scaffold template

### PR DESCRIPTION
**Pull request by asanehisa**

### CODE REVIEW

File: packages/pages/src/scaffold/template/generate.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      inactive: "Static",
+    },
+    {
+      type: (prev) => (prev ? "select" : null),
+      name: "entityScope",
+      message:
+        "How would you like you to define the entity scope for your template?",
+      choices: [
+        { title: "Entity Type", value: "entityTypes" },
+        { title: "Static", value: "static" },
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/template/generate.ts:42]: Static isn't a scope option. Aaron wrote "Static Filter" but he fixed after I commented. It should be "Saved Filter".

File: packages/pages/src/scaffold/template/generate.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  ];
+
+  const response = await prompts(questions);
+
+  // TODO (SUMO-5251): handle generating VE templates
+  // TODO (SUMO-5252): handle generating non-VE templates
+};
+
+// Returns true if templateName can be formatted into valid filename and that filename isn't being used.
+const validateTemplateName = (
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/template/generate.ts:80]: Don't we already have code that does this we can reuse?

File: packages/pages/src/scaffold/template/generate.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    formattedFileName
+  );
+  if (fs.existsSync(templatePath)) {
+    return false;
+  }
+
+  return true;
+};
+
+const formatFileName = (templateName: string): string => {
```
Comment (mkilpatrick) [inline packages/pages/src/scaffold/template/generate.ts:102]: Do we have this already too? If not, add docs for what it does.

Comment (asanehisa) [reply packages/pages/src/scaffold/template/generate.ts:102]: The code for it is in alpha unfortunately.

File: packages/pages/src/scaffold/template/generate.ts (hunk 2 above)
Comment (asanehisa) [reply packages/pages/src/scaffold/template/generate.ts:80]: Not quite, modules has a decent amount of diff naming rules than templates

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**